### PR TITLE
Bug 1182184 - Entering text and canceling on about:home doesn't clear URL text

### PR DESCRIPTION
**Pull request by bkmunar**
1. added delegate functions to detect if we're on the home page or not
2. if on the home page and we cancel, clear the editTextField

### CODE REVIEW

**Comment by fluffyemily** [comment]
This is a bit over complicated.
When you cancel a edit in the location view, the textFieldDidEndEditing is called. This calls highlightDomain, which sets the text field text back to what it was before. However, if the URL is empty, it does not do anything, leaving the text the way it was.

There are 2 solutions:

1. amend highlightDomain to set the text to nil if the URL is empty

OR

2. remove highlightDomain call from textFieldDidEndEditing, and then, in cancel(), after the check to see if the textField.text is empty, test whether or not we have a URL and call highlightDomain or set url to nil depending on the result.

I suggest doing #2, because highlightDomain is called inside didSet for url, which means when we actually navigate to a new URL, highlightDomain is called twice, unneccessarily. This way, highlightDomain is only called once, when the URL is set, and on cancel the correct value is set in cancel()

Make sense?

**Comment by bkmunar** [comment]
Ahhhhh ok I see, I was super confused as to why I couldn't find a better way to do this in the first place haha thanks for clarifying!!

**Comment by bkmunar** [comment]
MUUUCHHH better patch, thanks for the suggestion! <3

**Comment by bkmunar** [comment]
lttm

File: Client/Frontend/Browser/BrowserLocationView.swift [hunk 1]
```diff
@@ -294,6 +294,12 @@ class BrowserLocationView : UIView, UIGestureRecognizerDelegate, UITextFieldDele
         if editTextField.text.isEmpty {
             editTextField.text = clearedText ?? ""
         }
+
+        if let url = self.url {
```
Comment (fluffyemily) [inline Client/Frontend/Browser/BrowserLocationView.swift]: I think this should be an else if otherwise you'll just overwrite what we've just done

**Comment by bkmunar** [comment]
better?

**Pull request merged**